pyproject.toml

```diff
@@ -25,6 +25,8 @@ filterwarnings = [
     "ignore:DataFrameGroupBy.apply operated on the grouping columns.:DeprecationWarning",
     # Ignore Pandas 2.2 warning on PyArrow. It might be reverted in a later release.
     "ignore:\\s*Pyarrow will become a required dependency of pandas.*:DeprecationWarning",
+    # sklearn 1.6 deprecation warning, currently raised in the library itself
```

Comment:
where? This shouldn't happen

Reply:
One failure example(there are 12, all in `TestLightGBM`): 

```
_________________________________________________ TestLightGBM.test_classifier[dart] __________________________________________________

self = <test_external.TestLightGBM object at 0x14dd4f130>
lgbm = <module 'lightgbm' from '/Users/tamara/miniconda3/envs/skops/lib/python3.10/site-packages/lightgbm/__init__.py'>
clf_data = (array([[ 6.94667648e-02, -3.79071266e+00,  2.50068731e+00,
         3.52484579e+00,  2.58769644e+00,  1.75433348e+00,...91277e+00]]), array([0, 3, 2, 1, 2, 1, 3, 0, 0, 2, 0, 3, 2, 0, 2, 1, 1, 1, 3, 0, 1, 3,
       0, 3, 2, 1, 1, 2, 3, 0]))
trusted = ['collections.OrderedDict', 'lightgbm.basic.Booster', 'lightgbm.sklearn.LGBMClassifier', 'lightgbm.sklearn.LGBMRegressor', 'lightgbm.sklearn.LGBMRanker', 'sklearn.preprocessing._label.LabelEncoder']
boosting_type = 'dart'

    @pytest.mark.parametrize("boosting_type", boosting_types)
    def test_classifier(self, lgbm, clf_data, trusted, boosting_type):
        kw = {}
        if boosting_type == "rf":
            kw["bagging_fraction"] = 0.5
            kw["bagging_freq"] = 2

        estimator = lgbm.LGBMClassifier(boosting_type=boosting_type, **kw)
        loaded = loads(dumps(estimator), trusted=trusted)
        assert_params_equal(estimator.get_params(), loaded.get_params())

        X, y = clf_data
>       estimator.fit(X, y)

skops/io/tests/test_external.py:101:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniconda3/envs/skops/lib/python3.10/site-packages/lightgbm/sklearn.py:1284: in fit
    super().fit(
../../miniconda3/envs/skops/lib/python3.10/site-packages/lightgbm/sklearn.py:865: in fit
    _X, _y = _LGBMCheckXY(X, y, accept_sparse=True, force_all_finite=False, ensure_min_samples=2)
../../miniconda3/envs/skops/lib/python3.10/site-packages/sklearn/utils/validation.py:1366: in check_X_y
    ensure_all_finite = _deprecate_force_all_finite(force_all_finite, ensure_all_finite)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

force_all_finite = False, ensure_all_finite = None

    def _deprecate_force_all_finite(force_all_finite, ensure_all_finite):
        """Helper to deprecate force_all_finite in favor of ensure_all_finite."""
        if force_all_finite != "deprecated":
>           warnings.warn(
                "'force_all_finite' was renamed to 'ensure_all_finite' in 1.6 and will be "
                "removed in 1.8.",
                FutureWarning,
            )
E           FutureWarning: 'force_all_finite' was renamed to 'ensure_all_finite' in 1.6 and will be removed in 1.8.

../../miniconda3/envs/skops/lib/python3.10/site-packages/sklearn/utils/deprecation.py:146: FutureWarning
```

Reply:
Wait this is in LightGBM, seems to be solved [here](https://github.com/microsoft/LightGBM/blame/master/python-package/lightgbm/sklearn.py#L1085) and other lines below. They haven't done a release thought. I update the comment and keep the warning suppression for now?

Reply:
addressed in https://github.com/skops-dev/skops/pull/447/commits/d9a163b833a8dd06c5f927b8431fe9498deea66f, I can remove it when we know there is a new release.